core/rawdb/accessors_chain.go

```diff
@@ -556,3 +563,46 @@ func DeleteCXReceipts(db DatabaseDeleter, shardID uint32, number uint64, hash co
 		utils.Logger().Error().Msg("Failed to delete cross shard tx receipts")
 	}
 }
+
+// ReadCXReceiptsProofUnspent check whether a CXReceiptsProof is unspent, true means not spent
+func ReadCXReceiptsProofUnspent(db DatabaseReader, shardID uint32, number uint64) (common.Hash, error) {
+	data, err := db.Get(cxReceiptUnspentKey(shardID, number))
+	if err != nil || len(data) == 0 {
+		return common.Hash{}, ctxerror.New("[ReadCXReceiptsProofUnspent] Cannot find the key", "shardID", shardID, "number", number).WithCause(err)
+	}
+	return common.BytesToHash(data), nil
+}
+
+// TryWriteCXReceiptsProofUnspent check whether a CXReceiptsProof is unspent, write to database if unspent
+func TryWriteCXReceiptsProofUnspent(dbr DatabaseReader, dbw DatabaseWriter, shardID uint32, number uint64, blockHash common.Hash) error {
```

Comment:
This is not used?

Reply:
It will be used to avoid double spent, this part is not written yet.